src/Symfony/Bridge/Twig/NodeVisitor/TranslationDefaultDomainNodeVisitor.php

```diff
@@ -55,13 +53,22 @@ public function enterNode(Node $node, Environment $env): Node
                 return $node;
             }
 
-            $name = class_exists(AssignContextVariable::class) ? new AssignContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new AssignNameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine());
-            $this->scope->set('domain', class_exists(ContextVariable::class) ? new ContextVariable(self::INTERNAL_VAR_NAME, $node->getTemplateLine()) : new NameExpression(self::INTERNAL_VAR_NAME, $node->getTemplateLine()));
-
             if (class_exists(Nodes::class)) {
+                $name = new LocalVariable(null, $node->getTemplateLine());
+                $this->scope->set('domain', $name);
+
                 return new SetNode(false, new Nodes([$name]), new Nodes([$node->getNode('expr')]), $node->getTemplateLine());
             }
 
+            $var = '__internal_trans_default_domain';
+
+            if (null !== $templateName = $node->getTemplateName()) {
```

Comment:
I wonder what we do when there is no template name? We could use an internal counter in this visitor and append it to the prefix instead maybe?

Reply:
if a counter can replace any hashing, then yes, that'd be nicer